Provide instructions in INSTALL.md for go modules users

INSTALL.md should include working instructions for how to use this library when using go modules.

I have been unable to accomplish this. To reproduce, save the following as `Dockerfile` in an empty directory then `docker build .`:

    FROM golang:1.12

    WORKDIR /src

    RUN go mod init example

    RUN go get \
    k8s.io/client-go@kubernetes-1.13.4 \
    k8s.io/apimachinery@kubernetes-1.13.4

    RUN wget -q https://raw.githubusercontent.com/kubernetes/client-go/master/examples/in-cluster-client-configuration/main.go

    RUN go mod tidy

    RUN go build main.go

Output of the step that fails:

    ...
    Step 7/7 : RUN go build main.go
     ---> Running in 5014f5103213
    go: finding k8s.io/api/admissionregistration/v1alpha1 latest
    go: finding k8s.io/api/admissionregistration latest
    go: finding k8s.io/api latest
    build command-line-arguments: cannot load k8s.io/api/admissionregistration/v1alpha1: cannot find module providing package k8s.io/api/admissionregistration/v1alpha1
    The command '/bin/sh -c go build main.go' returned a non-zero code: 1

Please update `INSTALL.md` with what the `go get ...` should be for this to work.

```
../../../go/pkg/mod/k8s.io/client-go@v10.0.0+incompatible/kubernetes/scheme/register.go:22:2: unknown import path "k8s.io/api/admissionregistration/v1alpha1": cannot find module providing package k8s.io/api/admissionregistration/v1alpha1
```

--- a/staging/src/k8s.io/client-go/INSTALL.md
+++ b/staging/src/k8s.io/client-go/INSTALL.md
@@ -143,6 +143,48 @@ After modifying, run `glide up -v` again to re-populate your /vendor directory.
 Optionally, Glide users can also use [`glide-vc`](https://github.com/sgotti/glide-vc)
 after running `glide up -v` to remove unused files from /vendor.
 
+### Go modules
+
+Dependency management tools are built into go 1.11+ in the form of [go modules](https://github.com/golang/go/wiki/Modules).
+
+If you are using go 1.11 or 1.12 and are working with a project located within `$GOPATH`,
+you must opt into using go modules:
+
+```sh
+export GO111MODULE=on
+```
+
+Ensure your project has a `go.mod` file defined at the root of your project.
+If you do not already have one, `go mod init` will create one for you:
+
+```sh
+go mod init
+```
+
+Indicate the matching versions of `k8s.io/client-go`, `k8s.io/api`, and `k8s.io/apimachinery` your project requires:
+
+```sh
+go get k8s.io/client-go@v10.0.0              # replace v10.0.0 with the required version (or use kubernetes-1.x.y tags if desired)
+go get k8s.io/api@kubernetes-1.13.4          # replace kubernetes-1.13.4 with the required version
+go get k8s.io/apimachinery@kubernetes-1.13.4 # replace kubernetes-1.13.4 with the required version
+```
+
+or you can edit your `go.mod` file directly to specify required versions:
+
+```
+require (
+	k8s.io/api kubernetes-1.13.4
+	k8s.io/apimachinery kubernetes-1.13.4
+	k8s.io/client-go v10.0.0
+)
+```
+
+You can now import and use the `k8s.io/client-go` APIs in your project.
+The next time you `go build`, `go test`, or `go run` your project,
+`k8s.io/client-go` and its dependencies will be downloaded (if needed),
+and detailed dependency version info will be added to your `go.mod` file.
+You can also run `go mod tidy` to do this directly.
+
 ### Dep (Not supported yet!)
 
 [dep](https://github.com/golang/dep) is an up-and-coming dependency management

